### PR TITLE
docs: improve the docs regarding the test references

### PR DESCRIPTION
**Pull request by apahim**
Let's add some extra information and some examples about the test
references.

Reference: https://trello.com/c/iJlNcBaf
Signed-off-by: Amador Pahim <apahim@redhat.com>

### CODE REVIEW

**Comment by apahim** [comment]
[improve the docs regarding the test references](https://trello.com/c/iJlNcBaf/930-improve-the-docs-regarding-the-test-references)

File: docs/source/GetStartedGuide.rst [hunk 1]
```diff
@@ -211,6 +211,75 @@ keep reading.
           positional arguments, therefore test names (in case of
           ``avocado run``)
 
+Test References
+---------------
```
Comment (ldoktor) [inline docs/source/GetStartedGuide.rst]: Well this is quite detailed explanation, I'd not bother someone starting with Avocado with it. What would you think about moving this to `ReferenceGuide` section?

Comment (ldoktor) [inline docs/source/GetStartedGuide.rst]: Actually it's already described there and the detailed explanation here only applies to `FileLoader`. So how about tweak it a bit and create a `FileLoader discovery` section in `Loaders` with all these neat details?

File: docs/source/GetStartedGuide.rst [hunk 2]
```diff
@@ -211,6 +211,75 @@ keep reading.
...
+
+A Test Reference is a string that can be resolved into (interpreted as)
+one or more tests by the Avocado Test Resolver. It is usually a path for
+a test file and it can be followed by a `:` and then a regular expression
```
Comment (ldoktor) [inline docs/source/GetStartedGuide.rst]: Well this is up to the loader plugin, `avocado-vt`, `external` does not use `path` and `SIMPLE` test also does not support regexps.

File: docs/source/GetStartedGuide.rst (hunk 1 above)
Comment (apahim) [inline docs/source/GetStartedGuide.rst]: Yes, makes sense.

File: docs/source/GetStartedGuide.rst (hunk 2 above)
Comment (apahim) [inline docs/source/GetStartedGuide.rst]: agreed

**Comment by apahim** [comment]
Sure.

**Pull request closed without merging**